### PR TITLE
Fixes #3377 Remove the FrameworkId in case of a specific error message from Mesos

### LINKED ISSUE #3377: Marathon should remove the FrameworkId for special mesos errors
See MESOS-2522

### PR DESCRIPTION
**Pull request by aquamatthias**

### CODE REVIEW

**Comment by aquamatthias** [comment]
@kolloch See https://mesosphere.atlassian.net/browse/CORE-243

**Comment by kolloch** [comment]
a test? Otherwise looks "good" if matching for specific human-readable strings can be called "good".

File: src/main/scala/mesosphere/marathon/MarathonScheduler.scala [hunk 1]
```diff
@@ -117,8 +117,11 @@ class MarathonScheduler @Inject() (
 
     // Currently, it's pretty hard to disambiguate this error from other causes of framework errors.
     // Watch MESOS-2522 which will add a reason field for framework errors to help with this.
-    // For now the frameworkId is never removed!
-    val removeFrameworkId = false
+    // For now the frameworkId is removed based on the error message.
+    val removeFrameworkId = message match {
+      case "Framework has been removed" => true
+      case unknown: String => false
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/MarathonScheduler.scala]: unknown can be `_`

**Comment by aquamatthias** [comment]
Added 2 test cases for known and unknown errors.

**Comment by kolloch** [comment]
Thanks :100:

**Pull request merged**